specs/carousel/carousel.md

```diff
@@ -0,0 +1,258 @@
+# Carousel
+
+## Overview
+
+As defined by the W3C:
+> A carousel [aka Slideshow, Image Rotator] presents a set of items, referred to as slides, by sequentially displaying a subset of one or more slides. Typically, one slide is displayed at a time, and users can activate a next or previous slide control that hides the current slide and "rotates" the next or previous slide into view. In some implementations, rotation automatically starts when the page loads, and it may also automatically stop once all the slides have been displayed. While a slide may contain any type of content, image carousels where each slide contains nothing more than a single image are common.
+>
+
+
+### Use Cases
+
+- Jeff is looking for an apartment and wants to see what the apartment looks like, so he clicks on the main image and a carousel appears allowing him to rotate through the images of the apartment.
+
+- Jess is checking out the latest news from her favorite website and quickly sees the main headlines of the day at the top of the webpage as they rotate headline by headline across the page.
+
+### Features
+
+- **Next and Previous Controls:** Simple buttons [aka flippers] that allow for control of displaying the next and previous slides.
+
+- **Slide Picker:** A control or group of controls that allows the specific choice of slide to display.
+
+- **Autoplay:** Allows the carousel to begin rotating through slides automatically after the carousel loads. Autoplay can be toggled on/off with a button, and automatically stops when any element in the carousel receives keyboard focus or is hovered over.
+
+- **Looping:** Allows the user to navigate from the first slide to the last if the previous flipper is activated. If on the last slide the user can navigate from the last slide to the first when the next flipper is activated.
+
+
+### Risks and Challenges
+
+*Notable risks or challenges associated with implementing the component. Would we need to make any breaking changes in order to achieve this component's goals?*
+
+### Prior Art/Examples
+- [FAST-DNA (React)](https://explore.fast.design/components/carousel)
+- [W3C](https://w3c.github.io/aria-practices/examples/carousel/carousel-1.html)
+
+---
+
+## Design
+
+### API
+
+*Component name:*
+- `fast-carousel`
+
+*Attributes:*
+- `autoplay` - boolean value to toggle if the carousel plays on load or not (defaults to false)
+- `autoplay-interval` - a number representing the autoplay interval in milliseconds
+- `loop` - boolean value to toggle if the carousel loops slides at the beginning or end (defaults to true)
+- `paused` - whether the rotation is paused or not
+- `aria-roledescription` - set to *carousel* for overall component, and set to *slide* for each slide
```

Comment:
This shounds like it does not need to be an attribute if it always needs to be set to carousel or slide